src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -619,14 +628,33 @@ export const handleGetMetadata: RequestHandler<
     ...createReqMeta(req),
   }
 
-  // Specific query.
-  if (submissionId) {
-    return getSubmissionMetadata(formId, submissionId)
-      .map((metadata) => {
-        return metadata
-          ? res.json({ metadata: [metadata], count: 1 })
-          : res.json({ metadata: [], count: 0 })
+  return (
+    // Step 1: Retrieve logged in user.
+    getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Check whether user has read permissions to form.
+        getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Read,
+        }),
+      )
+      // Step 3: Check whether form is encrypt mode.
+      .andThen(checkFormIsEncryptMode)
+      // Step 4: Retrieve submission metadata.
+      .andThen<SubmissionMetadataList, DatabaseError>(() => {
```

Comment:
oh wow i didn't know you could do this. this is just for cleanliness so that on the `map`, we can hover and see that it's `SubmissionMetadataList` instead of a long type right

Reply:
no la, sometimes neverthrow not smart enough to deduce what it is from the inner block, so this is more akin to type casting the returnable types

Reply:
actually you don't need this casting right? I tried removing it and it works fine

Reply:
I could have sworn the types were `unknown` before... Nonetheless fixed in 0d80908